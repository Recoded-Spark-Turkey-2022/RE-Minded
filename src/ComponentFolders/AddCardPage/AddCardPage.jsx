import { React, useState, useMemo } from 'react';
import Select from 'react-select';
import countryList from 'react-select-country-list';
import Card1 from './Images/TopCard.svg';
import Card2 from './Images/BottomCard.svg';

function AddCard() {
  const [cardNumber, setCardNumber] = useState('');
  const [expiryDate, setExpiryDate] = useState('');
  const [cvv, setCvv] = useState('');
  const [nameOnCard, setNameOnCard] = useState('');
  const [zipCode, setZipCode] = useState('');
  const [city, setCity] = useState('');
  const [address, setAddress] = useState('');

  const [value, setValue] = useState('');
  const options = useMemo(() => countryList().getData(), []);

  const changeHandler = () => {
    setValue(value);
  };

  return (
    <div className="p-20 font-poppins">
      <h1 className="text-3xl pb-5 md:text-4xl lg:text-5xl">
        ADD CARD DETAILS
      </h1>
      <h2 className="pb-8 lg:pb-0 text-lg md:text-xl lg:text-2xl opacity-50">
        Please make sure all of the info you enter are the same as your
        registration info.
      </h2>
      <div className="flex flex-col lg:flex-row lg:space-x-96">
        <div className="text-md md:text-lg lg:text-xl items-center flex  flex-col md:flex-col lg:flex-row lg:space-x-52  ">
          <form className="w-full  opacity-50 pt-4 ">
            <label
              htmlFor="card type"
              className="block font-medium text-gray-700 mb-2"
            >
              Supported Card types
            </label>
            <div className="flex  pb-6 text-Buttons">
              <div className="text-cyan-200 h-12 px-1 py-2  border-2 border-r-0 w-full  text-center rounded-lg rounded-r-none">
                Visa
              </div>
              <div className=" h-12 px-1 py-2  border-2 w-full  text-center rounded-lg rounded-l-none">
                MasterCard
              </div>
            </div>

            <label
              htmlFor="card number"
              className="block font-medium text-gray-700 mb-2"
            >
              Card Number
            </label>
            <input
              className="w-full border border-gray-400 p-2 rounded-lg"
              type="text"
              value={cardNumber}
              onChange={(e) => setCardNumber(e.target.value)}
            />
            <div className="flex  space-x-7">
              <div>
                <label
                  htmlFor="expiry date"
                  className="block font-medium text-gray-700 mb-2 mt-4 "
                >
                  Expiry Date
                </label>
                <input
                  className="w-full border border-gray-400 p-2 rounded-lg"
                  type="text"
                  placeholder='MM/YY'
                  value={expiryDate}
                  onChange={(e) => setExpiryDate(e.target.value)}
                />
              </div>
              <div>
                <label
                  htmlFor="cvv"
                  className="block font-medium text-gray-700 mb-2 mt-4"
                >
                  CVV Code
                </label>
                <input
                  className="w-full border border-gray-400 p-2 rounded-lg"
                  type="text"
                  value={cvv}
                  placeholder="***"
                  onChange={(e) => setCvv(e.target.value)}
                />
              </div>
            </div>
            <label
              htmlFor="name on card"
              className="block font-medium text-gray-700 mb-2 mt-4"
            >
              Name on Card
            </label>
            <input
              className="w-full border border-gray-400 p-2 rounded-lg"
              type="text"
              value={nameOnCard}
              onChange={(e) => setNameOnCard(e.target.value)}
            />
          </form>
          <form className="w-full   opacity-50">
            <label
              htmlFor="country"
              className="block font-medium text-gray-700 mb-2 mt-4 "
            >
              Country
            </label>
            <Select
              className="pb-4"
              options={options}
              value={value}
              onChange={changeHandler}
            />
            <label
              htmlFor="zip code"
              className="block font-medium text-gray-700 mb-2 mt-4"
            >
              ZIP Code
            </label>
            <input
              className="w-full border border-gray-400 p-2 rounded-lg"
              type="text"
              value={zipCode}
              onChange={(e) => setZipCode(e.target.value)}
            />
            <label
              htmlFor="city"
              className="block font-medium text-gray-700 mb-2 mt-4"
            >
              City
            </label>
            <input
              className="w-full border border-gray-400 p-2 rounded-lg"
              type="text"
              value={city}
              onChange={(e) => setCity(e.target.value)}
            />
            <label
              htmlFor="address"
              className="block font-medium text-gray-700 mb-2 mt-4"
            >
              Address
            </label>
            <input
              className="w-full border border-gray-400 p-2 rounded-lg"
              type="text"
              value={address}
              onChange={(e) => setAddress(e.target.value)}
            />
          </form>

          <div className="pt-12 w-full flex flex-col justify-center">
            <img src={Card1} alt="TopCard" />
            <img src={Card2} alt="BottomCard" />
          </div>
        </div>
      </div>
      <div className="pt-12 lg:pt-0 text-md md:text-lg lg:text-xl flex justify-center lg:justify-start ">
        <button
          type="button"
          className="rounded-md box-border p-2 transition-all duration-250 bg-Buttons"
        >
          ADD CARD
        </button>
      </div>
    </div>
  );
}

export default AddCard;
