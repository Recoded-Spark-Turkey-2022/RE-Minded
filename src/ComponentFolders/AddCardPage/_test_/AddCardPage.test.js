import React from 'react';
import renderer from 'react-test-renderer';
import AddCardPage from '../AddCardPage';

test('Add Card Page Page snapshot test', () => {
  const component = renderer.create(<AddCardPage />);

  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
