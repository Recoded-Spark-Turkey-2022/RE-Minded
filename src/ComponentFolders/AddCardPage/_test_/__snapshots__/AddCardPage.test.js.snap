// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add Card Page Page snapshot test 1`] = `
<div
  className="p-20 font-poppins"
>
  <h1
    className="text-3xl pb-5 md:text-4xl lg:text-5xl"
  >
    ADD CARD DETAILS
  </h1>
  <h2
    className="pb-8 lg:pb-0 text-lg md:text-xl lg:text-2xl opacity-50"
  >
    Please make sure all of the info you enter are the same as your registration info.
  </h2>
  <div
    className="flex flex-col lg:flex-row lg:space-x-96"
  >
    <div
      className="text-md md:text-lg lg:text-xl items-center flex  flex-col md:flex-col lg:flex-row lg:space-x-52  "
    >
      <form
        className="w-full  opacity-50 pt-4 "
      >
        <label
          className="block font-medium text-gray-700 mb-2"
          htmlFor="card type"
        >
          Supported Card types
        </label>
        <div
          className="flex  pb-6 text-Buttons"
        >
          <div
            className="text-cyan-200 h-12 px-1 py-2  border-2 border-r-0 w-full  text-center rounded-lg rounded-r-none"
          >
            Visa
          </div>
          <div
            className=" h-12 px-1 py-2  border-2 w-full  text-center rounded-lg rounded-l-none"
          >
            MasterCard
          </div>
        </div>
        <label
          className="block font-medium text-gray-700 mb-2"
          htmlFor="card number"
        >
          Card Number
        </label>
        <input
          className="w-full border border-gray-400 p-2 rounded-lg"
          onChange={[Function]}
          type="text"
          value=""
        />
        <div
          className="flex  space-x-7"
        >
          <div>
            <label
              className="block font-medium text-gray-700 mb-2 mt-4 "
              htmlFor="expiry date"
            >
              Expiry Date
            </label>
            <input
              className="w-full border border-gray-400 p-2 rounded-lg"
              onChange={[Function]}
              placeholder="MM/YY"
              type="text"
              value=""
            />
          </div>
          <div>
            <label
              className="block font-medium text-gray-700 mb-2 mt-4"
              htmlFor="cvv"
            >
              CVV Code
            </label>
            <input
              className="w-full border border-gray-400 p-2 rounded-lg"
              onChange={[Function]}
              placeholder="***"
              type="text"
              value=""
            />
          </div>
        </div>
        <label
          className="block font-medium text-gray-700 mb-2 mt-4"
          htmlFor="name on card"
        >
          Name on Card
        </label>
        <input
          className="w-full border border-gray-400 p-2 rounded-lg"
          onChange={[Function]}
          type="text"
          value=""
        />
      </form>
      <form
        className="w-full   opacity-50"
      >
        <label
          className="block font-medium text-gray-700 mb-2 mt-4 "
          htmlFor="country"
        >
          Country
        </label>
        <div
          className="pb-4 css-b62m3t-container"
          onKeyDown={[Function]}
        >
          <span
            className="css-1f43avz-a11yText-A11yText"
            id="react-select-2-live-region"
          />
          <span
            aria-atomic="false"
            aria-live="polite"
            aria-relevant="additions text"
            className="css-1f43avz-a11yText-A11yText"
          />
          <div
            className=" css-13cymwt-control"
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
          >
            <div
              className=" css-1fdsijx-ValueContainer"
            >
              <div
                className=" css-1jqq78o-placeholder"
                id="react-select-2-placeholder"
              >
                Select...
              </div>
              <div
                className=" css-qbdosj-Input"
                data-value=""
              >
                <input
                  aria-autocomplete="list"
                  aria-describedby="react-select-2-placeholder"
                  aria-expanded={false}
                  aria-haspopup={true}
                  autoCapitalize="none"
                  autoComplete="off"
                  autoCorrect="off"
                  className=""
                  disabled={false}
                  id="react-select-2-input"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  role="combobox"
                  spellCheck="false"
                  style={
                    Object {
                      "background": 0,
                      "border": 0,
                      "color": "inherit",
                      "font": "inherit",
                      "gridArea": "1 / 2",
                      "label": "input",
                      "margin": 0,
                      "minWidth": "2px",
                      "opacity": 1,
                      "outline": 0,
                      "padding": 0,
                      "width": "100%",
                    }
                  }
                  tabIndex={0}
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              className=" css-1hb7zxy-IndicatorsContainer"
            >
              <span
                className=" css-1u9des2-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                className=" css-1xc3v61-indicatorContainer"
                onMouseDown={[Function]}
                onTouchEnd={[Function]}
              >
                <svg
                  aria-hidden="true"
                  className="css-tj5bde-Svg"
                  focusable="false"
                  height={20}
                  viewBox="0 0 20 20"
                  width={20}
                >
                  <path
                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
        <label
          className="block font-medium text-gray-700 mb-2 mt-4"
          htmlFor="zip code"
        >
          ZIP Code
        </label>
        <input
          className="w-full border border-gray-400 p-2 rounded-lg"
          onChange={[Function]}
          type="text"
          value=""
        />
        <label
          className="block font-medium text-gray-700 mb-2 mt-4"
          htmlFor="city"
        >
          City
        </label>
        <input
          className="w-full border border-gray-400 p-2 rounded-lg"
          onChange={[Function]}
          type="text"
          value=""
        />
        <label
          className="block font-medium text-gray-700 mb-2 mt-4"
          htmlFor="address"
        >
          Address
        </label>
        <input
          className="w-full border border-gray-400 p-2 rounded-lg"
          onChange={[Function]}
          type="text"
          value=""
        />
      </form>
      <div
        className="pt-12 w-full flex flex-col justify-center"
      >
        <img
          alt="TopCard"
          src="TopCard.svg"
        />
        <img
          alt="BottomCard"
          src="BottomCard.svg"
        />
      </div>
    </div>
  </div>
  <div
    className="pt-12 lg:pt-0 text-md md:text-lg lg:text-xl flex justify-center lg:justify-start "
  >
    <button
      className="rounded-md box-border p-2 transition-all duration-250 bg-Buttons"
      type="button"
    >
      ADD CARD
    </button>
  </div>
</div>
`;
